lib/ssl/test/tls_1_3_version_SUITE.erl

```diff
@@ -263,3 +277,92 @@ tls12_client_tls_server(Config) when is_list(Config) ->
+                                                  {from, self()},
+                                                  {mfa, {ssl_test_lib, send_recv_result_active, []}},
+                                                  {options,
+                                                   [{versions, ['tlsv1.2']}, {reuse_sessions, save}| ClientOpts]}]),
+    Server ! listen,
+    {ok,[{session_id, SessionId}, {session_data, SessData}]} = ssl:connection_information(CSock, [session_id, session_data]),
+    {_Client1, CSock1}  = ssl_test_lib:start_client([return_socket,
+                                                     {node, ClientNode}, {port, Port},
+                                                     {host, Hostname},
+                                                     {from, self()},
+                                                     {mfa, {ssl_test_lib, send_recv_result_active, []}},
+                                                     {options,
+                                                      [{versions, ['tlsv1.3', 'tlsv1.2']},
+                                                       {middlebox_comp_mode, true},
+                                                       {reuse_session, {SessionId, SessData}} | ClientOpts]}]),
+    {ok,[{session_id, SessionId}]}  = ssl:connection_information(CSock1, [session_id]).
+
+%%--------------------------------------------------------------------
+%% Internal functions and callbacks -----------------------------------
+%%--------------------------------------------------------------------
+
+middlebox_test(Mode, Expected, ClientOpts, ServerOpts, Config) ->
+    {ClientNode, ServerNode, Hostname} = ssl_test_lib:run_where(Config),
+    Server = ssl_test_lib:start_server([{node, ServerNode}, {port, 0},
+					{from, self()},
+                                        {mfa, {?MODULE, check_session_id, [Expected]}},
+                                        {options, ServerOpts}]),
+    Port = ssl_test_lib:inet_port(Server),
+    Client = ssl_test_lib:start_client([{node, ClientNode}, {port, Port},
+					{host, Hostname},
+                                        {from, self()},
+                                        {mfa, {?MODULE, check_session_id, [Expected]}},
+                                        {options,
+                                         [{middlebox_comp_mode, Mode}| ClientOpts]}]),
+    ssl_test_lib:check_result(Server, ok, Client, ok).
+
+check_session_id(Socket, empty) ->
+   {ok, [{session_id, SessionId}]} = ssl:connection_information(Socket, [session_id]),
+    case SessionId of
+        <<>> ->
+            ok;
+        _ ->
+            {nok, {{expected, <<>>}, {got, SessionId}}}
+    end;
+check_session_id(Socket, not_empty) ->
+    {ok, [{session_id, SessionId}]} = ssl:connection_information(Socket, [session_id]),
+    case SessionId of
+        <<>> ->
+            {nok, {{expected, not_empty}, {got, SessionId}}};
+        _ ->
+            ok
+    end.
```

Comment:
The two clauses are almost similar, may I propose something like this:
```suggestion
check_session_id(Socket, Expected) -> % Expected is empty | not_empty
   {ok, [{session_id, SessionId}]} = ssl:connection_information(Socket, [session_id]),
    case {Expected, SessionId} of
        {empty, <<>>} ->
            ok;
        {not_empty, Bin} when binary_size(Bin) > 0 -> 
            ok;
        {Exp, _} when Exp == empty; Exp == not_empty ->    
           {nok, {{expected, Exp}, {got, SessionId}}}
    end.
```

Reply:
Sorry got it wrong with use of `binary_size` which should be `byte_size`.

Reply:
No worries, I used SessionId =/= <<>>  instead. We do not need to have a type check here we just need to verify that we did not get an empty id!